common/src/main/java/io/netty/util/concurrent/SingleThreadEventExecutor.java

```diff
@@ -942,13 +963,15 @@ public void run() {
                             // See https://github.com/netty/netty/issues/6596.
                             FastThreadLocal.removeAll();
 
-                            STATE_UPDATER.set(SingleThreadEventExecutor.this, ST_TERMINATED);
-                            threadLock.release();
-                            if (!taskQueue.isEmpty()) {
-                                if (logger.isWarnEnabled()) {
-                                    logger.warn("An event executor terminated with " +
-                                            "non-empty task queue (" + taskQueue.size() + ')');
-                                }
+                            thread = nonVolatileThread = null;
+
+                            synchronized (threadLock) {
+                                STATE_UPDATER.set(SingleThreadEventExecutor.this, ST_TERMINATED);
```

Comment:
@njhill why does this need to be in the synchronized block ? I think you can move it out

Reply:
@normanmaurer I'm not sure it really makes any difference here but will do :)